src/app/service/item.service.ts

```diff
@@ -0,0 +1,87 @@
+// item.service.ts
+import { Injectable, inject } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Item } from '../models/item';
+import { BehaviorSubject, firstValueFrom } from 'rxjs';
+import { cartItem } from '../models/cartItem';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ItemService {
+  private fruitApiUrl =
```

Comment:
These are very similar so maybe refactor into one variable and append the filter type when needed